### PR TITLE
update contact footer to match design

### PR DESCRIPTION
**Pull request by aaaaargZombies**
The shortcode takes a message as the `.Inner` value and retrieves the email address from config.
It uses grid to accurately overlay the email address and image but does not use BEM.

Sidenote, Sam and Luke have said they prefer bold to black font-weight so I've left that as is for the moment.

### CODE REVIEW

**Comment by SamWhited** [comment]
I know I changed your initial design here, but I think we need to rethink the link in this. I added a slight background because otherwise it is unreadable to me against the green (or even the darker dove).

My accessibility tools indicate that the contrast is too low. I haven't tried, but I believe the white/lime contrast will be so low that increasing the font size won't allow us to be within AAA or even AA guidelines, so the background or changing the font color were the only options I could think of.

**Comment by aaaaargZombies** [comment]
Yes, you're right.

I'll think this will need a rethink then because the current main branch feels really out of keeping with the rest of the page.

**Comment by SamWhited** [comment]
I like the pill box; it feels a bit out of place because the rest of the site is kind of square, but the idea is really cool (and I'm not against leaving it as is, it looks good and I am certainly not the design person in the co-op).

File: config.toml [hunk 1]
```diff
@@ -18,4 +18,4 @@ better software for a better world."""
 [params.contact]
 
 org   = "Willow Bark Co-op"
-email = "hello@example.net"
```
Comment (SamWhited) [inline config.toml]: Unrelated change, can this be removed from this PR? (although, "oops, good catch, I thought this was fixed already!")

File: themes/salix/layouts/shortcodes/contact.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<section class="contact">
+	<div>
+		<h2 id="contact">Contact</h2>
+		<div>
+			<p>
+				{{ .Inner }}
+			</p>
+			<a href="mailto:{{ .Site.Params.contact.email }}">
+				{{ .Site.Params.contact.email }}
```
Comment (SamWhited) [inline themes/salix/layouts/shortcodes/contact.html:9]: I know I'm being super nitpicky here in terms of best practices, but I think *this* should be the shortcode (ie. "email-pillbox" or something), not the entire section including some of its content. Ie. let's just make this a section with an ID, in the CSS apply the pigeon as a background on that section ID, and in the markdown content add the "h2" and the email shortcode.

File: config.toml (hunk 1 above)
Comment (aaaaargZombies) [reply config.toml]: sure it's now in #19

File: themes/salix/layouts/shortcodes/contact.html (hunk 2 above)
Comment (aaaaargZombies) [reply themes/salix/layouts/shortcodes/contact.html:9]: > I like the pill box; it feels a bit out of place because the rest of the site is kind of square, but the idea is really cool (and I'm not against leaving it as is, it looks good and I am certainly not the design person in the co-op).

yeah it was a bit of a stab in the dark, I should probably do a bit of visual research to see how other people have solved this issue then makes some variations in Figma. I just wanted something that would make it accessible while changing the design as little as possible for the mean time.

**Comment by aaaaargZombies** [comment]
> I like the pill box; it feels a bit out of place because the rest of the site is kind of square, but the idea is really cool (and I'm not against leaving it as is, it looks good and I am certainly not the design person in the co-op).

yeah it was a bit of a stab in the dark, I should probably do a bit of visual research to see how other people have solved this issue then makes some variations in Figma. I just wanted something that would make it accessible while changing the design as little as possible for the mean time.

File: themes/salix/layouts/shortcodes/contact.html (hunk 2 above)
Comment (aaaaargZombies) [reply themes/salix/layouts/shortcodes/contact.html:9]: I'll follow this up, I'm concerned about being able to accurately place the image but I'll have an experiment and see if I can make it work.

Comment (SamWhited) [reply themes/salix/layouts/shortcodes/contact.html:9]: Generally speaking I think web design needs to operate within the constraints of web development. If we can't match the design exactly 1:1 that's fine. Changes will likely have to be made for eg. accessibility or to follow best practices and that's okay. Absolute positioning isn't one of the things we can do with web design, so might as well just turn off repeat, position it as well as we can, and call it a day IMO :)

**Comment by aaaaargZombies** [comment]
OK I think hope this is more what you were thinking in relation to shortcode use. I realize now I had misunderstood a couple of things about Hugo. I thought you had made `section` and `div` shortcodes because it wasn't possible to pass HTML in from the markdown file. But after some faffing I see that you made the `div` shortcode so that you could apply markdown after having passed HTML to `flexsection`.

The `flexsection` shortcode ended up working very well but is now maybe ill named as the inner uses grid layout.

I spent a while reading WCAG to try and get some clarity on what is best practice for decorative images. It seems that `aria-hidden="ture"` still satisfies success criterion 1.1.1  and will likely become a  recommended technique for use with [icon fonts](https://www.w3.org/WAI/GL/wiki/Using_aria-hidden%3Dtrue_on_an_icon_font_that_AT_should_ignore), I assume this will also apply to images placed with grid which was maybe less common place when WCAG 2.1 was being put together.

**Comment by SamWhited** [comment]
Apologies again for letting this go so long that a rebase is necessary. Re-reviewing this looks great and my only concerns are nit-picks:

- I still think the image needs to be a background. This will be semantically correct and not require workarounds for screen readers and the like. I wouldn't block on this though.
- On a small screen (I used the iPhone XS preview in Firefox) the email link has a lot of margin separating it from the rest of the text (I think it just needs to be inside the same paragraph)

Looks great though!

**Comment by aaaaargZombies** [comment]
1. I feel quite strongly about not writing off a huge part of CSS layout so if you're not going to block then lets move forward as is for now.
2. This was intentional as originally it was supposed to use the bird as a background. I think this is a reason we need a visual review stage before moving to code.

My thoughts at the moment are that it would be better to merge some stuff in just so it looks more finished if we're going to start speaking to potential clients. We can then open some of these things as issues and reduce the scale of the pull requests.

Thoughts?

**Comment by SamWhited** [comment]
> I feel quite strongly about not writing off a huge part of CSS layout

Can you expand on this more, I wasn't sure what this meant, sorry. As I said, I wouldn't block on this particular point but in my experience trying to make images backgrounds leads to issues later that you don't have with an actual background and a few CSS properties.

> This was intentional as originally it was supposed to use the bird as a background

It doesn't anymore though, correct? The bird background probably shouldn't be on a link or a paragraph, but on the section or other wrapper div. If it's not required now, let's just move this into the paragraph, it's definitely part of the same text and breaking it out will just be confusing for layout engines and screen readers and the like.

**Comment by SamWhited** [comment]
To make the email bit easier to discuss and so that we can get some of this merged incrementally instead of the whole thing blocking on a few nit picks, I cherry picked 77027b6e30cd57fb96a371d503c11971317defcd over and removed everything except the email shortcode and made a new PR we can use: https://codeberg.org/willowbark/web/pulls/12

Normally I wouldn't as it will cause conflicts, but since this needs to be rebased anyways I figured we might as well pull out the less controversial parts (I think?)

If we're following best practices then someone else should review it since I tweaked it a bit and made the PR and you made the actual commit; but in the interest of expediency I think it's fine if either of us merges it (assuming you agree, if not naturally we can do something else)

**Comment by SamWhited** [comment]
Sorry, last thing: I just realized that you want to do the layout this way because otherwise it's easier for the bird to accidentally overlap the navy text and make it unreadable. That's fair, and I will have to think about that for a bit. My intuition is that it's better to design it in such a way that the bird doesn't have to be absolutely positioned so that it doesn't overlap the text, but I'm not sure.

**Comment by aaaaargZombies** [comment]
>> I feel quite strongly about not writing off a huge part of CSS layout

> Can you expand on this more, I wasn't sure what this meant, sorry.

TLDR: grid.

I'll give it my best shot, essentially grid allows us to describe the relationship between the images,text and white-space on the page in ways that haven't been possible on the web previously. Those things do matter to end users and we're going to see people designing websites in ways that have been limited to print in the past. That's going to mean we need to place decorative images into the DOM.

If you have any time this [talk](https://www.youtube.com/watch?v=t0b3uBoDkBs) is interesting, she goes through different stages of CSS history and talks a bit about magazines, posters and film.

Code from some of her examples is [here](https://labs.jensimmons.com/)

**Comment by SamWhited** [comment]
Oh sure, grid is great and I'm all for using it where appropriate, I just don't think a background image is necessarily a good use for it. That being said, I couldn't figure out a good way to do this with an actual background image, and I know this is something most people don't have any qualms about.
